hedge.go

```diff
@@ -0,0 +1,97 @@
+package parallel
+
+import (
+	"context"
+	"time"
+)
+
+type HedgedRequestConfig struct {
+	delay             time.Duration // time to wait before issuing hedged requests
+	numHedgedRequests int           // the maximum permitted number of outstanding hedged requests
+}
+
+func (h *HedgedRequestConfig) Apply(opts ...HedgedRequestOpt) {
+	for _, opt := range opts {
+		opt(h)
+	}
+}
+
+type HedgedRequestOpt func(config *HedgedRequestConfig)
+
+func WithDelay(delay time.Duration) HedgedRequestOpt {
+	return func(config *HedgedRequestConfig) {
+		config.delay = delay
+	}
+}
+
+func WithNumHedgedRequests(numHedgedRequests int) HedgedRequestOpt {
+	return func(config *HedgedRequestConfig) {
+		config.numHedgedRequests = numHedgedRequests
+	}
+}
+
+func HedgedRequest[T any](
+	ctx context.Context,
+	requester func(context.Context) (T, error),
+	opts ...HedgedRequestOpt,
+) (T, error) {
+	cfg := HedgedRequestConfig{
+		delay:             50 * time.Millisecond,
+		numHedgedRequests: 2,
+	}
+	cfg.Apply(opts...)
+
+	hedgeSignal := make(chan struct{}) // closed when hedged requests should fire
+	responses := make(chan T)          // unbuffered, we only expect one response
+	ctx, cancel := context.WithCancelCause(ctx)
+
+	group := ErrGroup(Limited(ctx, cfg.numHedgedRequests+1))
```

Comment:
because we don't send an unbounded number of tasks to the group, it can be `Unlimited` for cheaper. A `Limited` group with N max parallelism that only ever receives N tasks doesn't do anything different.